logstash-core/lib/logstash/bootstrap_check/bad_java.rb

```diff
@@ -0,0 +1,17 @@
+# encoding: utf-8
+require "logstash/util"
+require "logstash/util/java_version"
+require "logstash/errors"
+
+module LogStash module BootstrapCheck
+  class BadJava
+    def self.check(settings)
+      # Exit on bad java versions
+      java_version = LogStash::Util::JavaVersion.version
+
+      if LogStash::Util::JavaVersion.bad_java_version?(java_version)
+        raise LogStash::BootstrapCheckError, "Java version 1.8.0 or later is required. (You are running: #{java_version})"
```

Comment:
I've moved the actual code that raise the exection inside the JavaVersion module and call instead `validate_java_version!`, this make all the logic encapsulated in the same file and the bootstrap_check is just a thin wrapper around that.